src/main/java/com/gregtechceu/gtceu/api/gui/widget/TankWidget.java

```diff
@@ -540,8 +540,11 @@ private int tryClickContainer(boolean isShiftKeyDown) {
                 FluidActionResult result = FluidUtil.tryEmptyContainer(currentStack, fluidTank, Integer.MAX_VALUE, null,
```

Comment:
I think we could actually solve this more elegantly by calculating the amount of empty space in the tank and passing that to `tryEmptyContainer` rather than using INT_MAX. 

If you follow through the FluidUtil steps, you can see that it first asks the bucket how much it can drain up to the max amount - which will be the full 1000 if we pass in INT_MAX. Later, when it actually *tries* to drain the bucket with the amount that was filled, it can't. This is why the simulation passes but the actual transfer fails.

Instead we can just tell FluidUtil how much we actually want to transfer.